src/Extension.php

```diff
@@ -599,6 +617,125 @@ public function rcp_membership_payment_profile_cancelled( $success, $gateway, $g
 		return true;
 	}
 
+	/**
+	 * Can update membership billing.
+	 *
+	 * @param bool $can_update        Whether or not billing can be updated.
+	 * @param int  $rcp_membership_id ID of the membership.
+	 * @return bool
+	 */
+	public function rcp_membership_can_update_billing( $can_update, $rcp_membership_id ) {
+		$subscriptions = \get_pronamic_subscriptions_by_source( 'rcp_membership', $rcp_membership_id );
+
+		if ( 0 !== count( $subscriptions ) ) {
+			$can_update = true;
+		}
+
+		return $can_update;
+	}
+
+	/**
+	 * Subscription action links HTML.
+	 *
+	 * @param string         $actions        Formatted HTML links.
+	 * @param array          $links          Links.
+	 * @param int            $user_id        Current user ID.
+	 * @param RCP_Membership $rcp_membership Membership object.
+	 * @return string
+	 */
+	public function rcp_subscription_details_actions( $actions, $links, $user_id, $rcp_membership ) {
+		$subscriptions = \get_pronamic_subscriptions_by_source( 'rcp_membership', $rcp_membership->get_id() );
+
+		if ( 0 === count( $subscriptions ) ) {
+			return $actions;
+		}
+
+		$subscription = reset( $subscriptions );
+
+		foreach ( $links as $link ) {
+			if ( ! \str_contains( $link, 'rcp_sub_details_update_card' ) ) {
+				continue;
+			}
+
+			$new_link = new WP_HTML_Tag_Processor( $link );
+
+			if ( $new_link->next_tag( 'a' ) ) {
+				$new_link->set_attribute( 'href', $subscription->get_mandate_selection_url() );
+			}
+
+			$actions = \str_replace( $link, $new_link, $actions );
+		}
+
+		return $actions;
+	}
+
+	/**
+	 * Maybe update membership gateway on subscription updates.
+	 *
+	 * @param int $post_id Subscription post ID.
+	 * @return void
+	 */
+	public function maybe_update_membership_gateway( $post_id ) {
+		$subscription = \get_pronamic_subscription( $post_id );
+
+		if ( null === $subscription ) {
+			return;
+		}
+
+		if ( 'rcp_membership' !== $subscription->get_source() ) {
+			return;
+		}
+
+		$rcp_membership = \rcp_get_membership( $subscription->get_source_id() );
+
+		if ( false === $rcp_membership ) {
+			return;
+		}
+
+		$payment_methods = [
+			$subscription->get_payment_method(),
+		];
+
+		$payments = $subscription->get_payments();
+
+		foreach ( $payments as $payment ) {
+			if ( 'subscription_payment_method_change' !== $payment->get_source() ) {
+				continue;
+			}
+
+			$payment_methods[] = $payment->get_payment_method();
+
+			break;
+		}
```

Comment:
This is not scalable, in theory hundreds of payments can be linked to a subscription.